### PR TITLE
Add additional tests for distributed tracing enabled

### PR DESCRIPTION
**Pull request by brettlangdon**
Pull out how we check if distributed tracing is enabled for an integration config into a helper that can be used by integrations and will match the logic we have in `activate_distributed_headers`

### CODE REVIEW

File: ddtrace/contrib/trace_utils.py [hunk 1]
```diff
@@ -165,6 +165,16 @@ def wrapper(wrapped, instance, args, kwargs):
     return with_mod
 
 
+def distributed_tracing_enabled(int_config, default=False):
+    # type: (IntegrationConfig, bool) -> bool
+    """Returns whether distributed tracing is enabled for this integration config"""
+    if "distributed_tracing_enabled" in int_config and int_config.distributed_tracing_enabled is not None:
+        return int_config.distributed_tracing_enabled
+    elif "distributed_tracing" in int_config and int_config.distributed_tracing is not None:
```
Comment (mabdinur) [inline ddtrace/contrib/trace_utils.py:173]: Why do we use distributed_tracing_enabled and distributed_tracing keys in int_config? Is this for backwards compatibility?

File: tests/tracer/test_trace_utils.py [hunk 2]
```diff
@@ -412,6 +412,56 @@ def test_bad_http_code(mock_log, span, int_config, val, bad):
...
+        ({"distributed_tracing_enabled": True, "distributed_tracing": None}, None, True),
+        ({"distributed_tracing_enabled": None, "distributed_tracing": True}, None, True),
+        ({"distributed_tracing_enabled": None, "distributed_tracing": None}, None, False),
+        ({"distributed_tracing_enabled": None, "distributed_tracing": None}, False, False),
+        ({"distributed_tracing_enabled": None, "distributed_tracing": None}, True, True),
+        ({"distributed_tracing_enabled": False, "distributed_tracing": True}, None, False),
+        ({"distributed_tracing_enabled": False, "distributed_tracing": False}, None, False),
+    ),
+)
+def test_distributed_tracing_enabled(int_config, props, default, expected):
```
Comment (mabdinur) [inline tests/tracer/test_trace_utils.py:454]: Are all these test cases necessary?

File: ddtrace/contrib/trace_utils.py (hunk 1 above)
Comment (brettlangdon) [reply ddtrace/contrib/trace_utils.py:173]: yup! for backwards compatibility reasons. we didn't unify integrations and messed up the config in the beginning.

File: tests/tracer/test_trace_utils.py (hunk 2 above)
Comment (brettlangdon) [reply tests/tracer/test_trace_utils.py:454]: probably not _all_ necessary, but the verbosity doesn't hurt, especially given their cost

**Review by Yun-Kim — COMMENTED** [review]
Should we add `distributed_tracing_enabled()` to be used in our integrations in this PR as well?

**Comment by brettlangdon** [comment]
> Should we add `distributed_tracing_enabled()` to be used in our integrations in this PR as well?

We use this in `activate_distributed_headers` which is used by most integrations.

File: ddtrace/contrib/trace_utils.py [hunk 3]
```diff
@@ -165,6 +165,16 @@ def wrapper(wrapped, instance, args, kwargs):
...
+        return int_config.distributed_tracing
+    return default
```
Comment (Yun-Kim) [inline ddtrace/contrib/trace_utils.py:175]:
```suggestion
    return int_config.get("distributed_tracing_enabled", int_config.get("distributed_tracing", False))
```
Can probably be simplified to this (and remove `default` as an argument)?

Comment (brettlangdon) [reply ddtrace/contrib/trace_utils.py:175]: now I cannot remember why I did it this way... 🤔

**Review by Kyle-Verhoog — DISMISSED** [review]
:+1: can we just add the motivation for the change in the PR description?

**Comment by brettlangdon** [comment]
> 👍 can we just add the motivation for the change in the PR description?

done

**Review by Kyle-Verhoog — APPROVED** [review]

File: ddtrace/contrib/trace_utils.py (hunk 3 above)
Comment (brettlangdon) [reply ddtrace/contrib/trace_utils.py:175]: ah, this is why... good thing I had these tests to remind myself

https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/10227/workflows/9d5a3317-0898-4ce0-ba19-9d75cf6d0c81/jobs/788436

if the value is `None` we want to ignore it and move on to the next possible one.

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Pull request merged**